### PR TITLE
handle 404 ES errors

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?186039
@czue 

cc @sravfeyn

### CODE REVIEW

File: corehq/pillows/user.py [hunk 1]
```diff
@@ -93,6 +94,9 @@ def get_fields_from_doc(self, doc):
     def _user_exists(self, user_id):
         return self.user_db.doc_exist(user_id)
 
+    def _user_indexed(self, user_id):
+        return doc_exists_in_es('users', user_id)
```
Comment (dimagimon) [inline corehq/pillows/user.py]: F821 undefined name 'doc_exists_in_es'

File: corehq/elastic.py [hunk 2]
```diff
@@ -37,6 +38,18 @@ def get_es(timeout=30):
                                     settings.ELASTICSEARCH_PORT),
                          timeout=timeout)
 
+
+def doc_exists_in_es(index, doc_id):
+    path = ES_URLS[index].replace('_search', doc_id)
```
Comment (czue) [inline corehq/elastic.py]: (just musing but this is a pretty funny/sketchy way to construct this url)

**Comment by czue** [comment]
:+1:

File: corehq/elastic.py (hunk 2 above)
Comment (snopoke) [inline corehq/elastic.py]: +1

**Pull request merged**